src/main/kotlin/Todo/ToDoList.kt

```diff
@@ -31,4 +37,9 @@ class ToDoList(private val filePath: String, private val fileIo: FileInterface,
             listOfTasks.size
         }
     }
+
+    private fun retrieveTaskData(jsonString: String): String {
+        val obj = jsonGenerator.resourceFromJson(jsonString, mutableMapOf<String, String>()::class)
+        return obj.getValue("task")
+    }
```

Comment:
Nitpick: Looks like you're parsing the request body here. I wonder if this should happen in the Action. 

My thought is that the ToDoList should not know about the HTTP request body structure. I think that the Action (or often controller) should be the layer that knows about the HTTP request.

Reply:
That was my original thought as well but would require me to inject `JsonGenerator` into the `CreateToDoAction`. So both the `CreateToDoAction` as well as the `ToDoList`, which is a dependency would have `JsonGenerator`. I believe removing `JsonGenerator` from `ToDoList`, and adding it to `CreateToDoAction`, and leaving it to the `CreateToDoAction` to create a `Task` after parsing the request body would make more sense. What are your thoughts?